arrow-avro/src/reader/record.rs

```diff
@@ -1421,6 +1849,24 @@ impl Skipper {
                 }
                 Ok(())
             }
+            Self::Union(encodings) => {
+                // Union tag must be ZigZag-decoded
+                let branch = buf.get_long()?;
+                if branch < 0 {
+                    return Err(ArrowError::ParseError(format!(
+                        "Negative union branch index {branch}"
+                    )));
+                }
+                let idx = branch as usize;
+                if let Some(encoding) = encodings.get_mut(idx) {
+                    encoding.skip(buf)
+                } else {
+                    Err(ArrowError::ParseError(format!(
+                        "Union branch index {idx} out of range for skipper ({} branches)",
+                        encodings.len()
+                    )))
+                }
```

Comment:
nit: Why not just call it `index` or `idx` from the start, and then:
```suggestion
                let Some(encoding) = encodings.get_mut(idx as usize) else {
                    return Err(ArrowError::ParseError(format!(
                        "Union branch index {idx} out of range for skipper ({} branches)",
                        encodings.len()
                    )));
                };
                encoding.skip(buf)
```